service/frontend/workflow_handler.go

```diff
@@ -874,6 +874,22 @@ func (wh *WorkflowHandler) PollWorkflowTaskQueue(ctx context.Context, request *w
 	pollerID := uuid.New()
 	childCtx := wh.registerOutstandingPollContext(ctx, pollerID, namespaceID.String())
 	defer wh.unregisterOutstandingPollContext(pollerID, namespaceID.String())
+
+	// route heartbeat to the matching service only if the request is valid (all validation checks passed)
+	if request.WorkerHeartbeat != nil {
+		_, err = wh.matchingClient.RecordWorkerHeartbeat(ctx, &matchingservice.RecordWorkerHeartbeatRequest{
```

Comment:
I don't like doing this synchronously, it'll add latency to every poll. can we just fire off a goroutine and do it in the background?